src/common/btree.c

```diff
@@ -2684,7 +2690,7 @@ btr_tx_delete(struct btr_context *tcx, void *args)
  *				args to handle special cases(if any)
  */
 int
-dbtree_delete(daos_handle_t toh, d_iov_t *key,
+dbtree_delete(daos_handle_t toh, dbtree_probe_opc_t opc, d_iov_t *key,
```

Comment:
I'm curious how will this 'opc' being used? Won't 'dbtree_delete()' always delete the specified key?

Reply:
just like probe, e.g. delete the key with equal or lower epoch

Reply:
this is required before we can have incarnation log: if we want to delete an object, all we can do is deleting the one with highest epoch, so we input EPOCH_MAX.

I think it is also useful for generic case, for example, we can easily delete the first/last one for integer key.